packages/contracts-bedrock/test/kontrol/proofs/utils/KontrolUtils.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Vm } from "forge-std/Vm.sol";
+import { Types } from "src/libraries/Types.sol";
+import { KontrolCheats } from "kontrol-cheatcodes/KontrolCheats.sol";
+
+contract GhostBytes {
+    bytes public ghostBytes;
+}
+
+contract GhostBytes10 {
+    bytes public ghostBytes0;
+    bytes public ghostBytes1;
+    bytes public ghostBytes2;
+    bytes public ghostBytes3;
+    bytes public ghostBytes4;
+    bytes public ghostBytes5;
+    bytes public ghostBytes6;
+    bytes public ghostBytes7;
+    bytes public ghostBytes8;
+    bytes public ghostBytes9;
+
+    function getGhostBytesArray() public view returns (bytes[] memory _arr) {
+        _arr = new bytes[](10);
+        _arr[0] = ghostBytes0;
+        _arr[1] = ghostBytes1;
+        _arr[2] = ghostBytes2;
+        _arr[3] = ghostBytes3;
+        _arr[4] = ghostBytes4;
+        _arr[5] = ghostBytes5;
+        _arr[6] = ghostBytes6;
+        _arr[7] = ghostBytes7;
+        _arr[8] = ghostBytes8;
+        _arr[9] = ghostBytes9;
+    }
+}
+
+/// @notice tests inheriting this contract cannot be run with forge
```

Comment:
Tangential: I'd be curious to hear the minimum set of features needed in forge so that the same tests can be run with both forge and kontrol

Reply:
Basically, supporting the KEVM cheatcodes that are not part of Foundry. We're currently using `fresUInt` and `symbolicStorage`, but it's unclear if these will be used in the final stage since we're using them for the symbolic workarounds.
So, just inheriting `KontrolUtils` will not actually make it incompatible with forge, but calling any of its functions will

Reply:
Got it, thanks. I'll make a note of this for now and keep it in mind :)